### DISCUSSION TITLE
RADdata2VCF is reference / alt format possible?

### DISCUSSION THREAD

**Discussion by relshire** [lvclark/polyRAD#41] [Q&A] [answered]
My analysis is using readTASSELGBSv2 to import the data. When using RADdata2VCF to generate the VCF file, it gives the following warning messages:

`RADdata2VCF(mydataPopStruct, file = "output.vcf", asSNPs = TRUE, hindhe = TRUE, contigs = mycontigs)`

`Warning messages:
1: In RADdata2VCF(mydataPopStruct, file = "ouptut.vcf",  :
  Reference allele not indicated.  Using the major allele for each locus.
2: In .Call(.make_vcf_geno, filename, fixed, names(geno), as.list(geno),  :
  converting NULL pointer to R NULL
`

I have done my best reading through the documentation. Even adding the contigs from the reference fasta file hoping that this would add the reference allele information. The same warning messages appear. At this point, I am left wondering if it is not possible to get the reference / alt format in this use case, or if I am just missing something.

Any help will be greatly appreciated!

Cheers,

Rob

**Comment by lvclark** [comment, marked as answer]
Hi Rob,

Generally the reference allele information gets added to the RADdata object by using the `refgenome` argument in the `VCF2RADdata` function.  If that is not how you are importing the data before genotype calling, then you could instead manually add the reference allele identity as the `object$locTable$Ref` column.

Best,

Lindsay

**Reply by relshire** [reply]
Thanks heaps Lindsay!
